src/commands/functions/functions-serve.ts

```diff
@@ -37,6 +40,15 @@ export const functionsServe = async (options: OptionValues, command: BaseCommand
     siteInfo,
   })
 
+  if (!options.offline) {
+    await setupAIGateway({ api, env, siteID: site.id, siteURL: siteUrl })
+  }
+
+  if (env.AI_GATEWAY?.value) {
+    process.env.AI_GATEWAY = env.AI_GATEWAY.value
```

Comment:
Same comment that I left on the `dev` command about manually injecting the variable into the process.